src/tests/util.rs

```diff
@@ -135,9 +135,10 @@ impl TestApp {
         use diesel::prelude::*;
 
         let user = self.db(|conn| {
-            let mut user = crate::new_user(username).create_or_update(conn).unwrap();
-            let email = "something@example.com";
-            user.email = Some(email.to_string());
+            let email = "cow@mammals.milk";
+            let user = crate::new_user(username)
+                .create_or_update(Some(email), conn)
```

Comment:
Aha, so the unique constraint that's causing the tests to fail is that here in the test setup you're creating a new user with an email set, then [lines 142-149 right after this](https://github.com/rust-lang/crates.io/pull/1891/files#diff-2bda8a5d16907d0b0ef38311d9dccb43R142-R149) the test setup is inserting the email into the table directly, which violates [the constraint we have that each user may only have one email](https://github.com/rust-lang/crates.io/blob/21e04f3028624485621d577359008560728a37d3/migrations/20170804200817_add_email_table/up.sql#L7).

To fix this, you can pass `None` to `create_or_update` here and the next part will take care of inserting the email (and setting it to verified, which is important for the tests).